slsReceiverSoftware/src/DataProcessor.cpp

```diff
@@ -28,7 +28,7 @@ DataProcessor::DataProcessor(int ind, detectorType dtype, Fifo* f,
 		bool* dsEnable, bool* gpEnable, uint32_t* dr,
 		uint32_t* freq, uint32_t* timer,
 		bool* fp, bool* act, bool* depaden, bool* sm,
-		int* ct, int* cdo, int* cad) :
+		std::vector <int> * cdl, int* cdo, int* cad) :
```

Comment:
Why do you use a pointer to a vector?

Reply:
So that if there are changes done to the vector in the tcp interface, it will reflect. The same with all the other pointers.